lib/modules/dosomething/dosomething_mbp/dosomething_mbp.admin.inc

```diff
@@ -9,11 +9,212 @@
+        '#default_value' => variable_get($template_var),
+        '#description' => 'The Mandrill template to use for Password Reset transactional messages.',
+        '#required' => TRUE,
+        '#weight' => 80,
+      );
+      $template_var = 'dosomething_mbp_campaign_signup_template';
+      $form['dosomething_mbp_templates']['dosomething_mbp_campaign_signup_template'] = array(
+        '#type' => 'select',
+        '#title' => t('Campaign Signup'),
+        '#empty_option' =>t('-- Select Mandrill Template --'),
+        '#options' => $mandrill_templates['select'],
+        '#default_value' => variable_get($template_var),
+        '#description' => 'The Mandrill template to use for Campaign Signup transactional messages.',
+        '#required' => TRUE,
+        '#weight' => 90,
+      );
+      $template_var = 'dosomething_mbp_campaign_group_signup_template';
+      $form['dosomething_mbp_templates']['dosomething_mbp_campaign_group_signup_template'] = array(
+        '#type' => 'select',
+        '#title' => t('Campaign Group Signup'),
+        '#empty_option' =>t('-- Select Mandrill Template --'),
+        '#options' => $mandrill_templates['select'],
+        '#default_value' => variable_get($template_var),
+        '#description' => 'The Mandrill template to use for Campaign Group Signup transactional messages.',
+        '#required' => TRUE,
+        '#weight' => 90,
+      );
+      $template_var = 'dosomething_mbp_campaign_reportback_template';
+      $form['dosomething_mbp_templates']['dosomething_mbp_campaign_reportback_template'] = array(
+        '#type' => 'select',
+        '#title' => t('Campaign Reportback'),
+        '#empty_option' =>t('-- Select Mandrill Template --'),
+        '#options' => $mandrill_templates['select'],
+        '#default_value' => variable_get($template_var),
+        '#description' => 'The Mandrill template to use for Campaign Reportback transactional messages.',
+        '#required' => TRUE,
+        '#weight' => 100,
+      );
+    }
+
+  }
+
   return system_settings_form($form);
 }
+
+/**
+ * Format template data for presentation based on returned values from Mandrill.
+ *
+ * @parm string $application_id
+ *   Defined in message_broker_producer, a unique identifier of the site.
+ */
+function dosomething_mbp_get_mandrill_template_options($application_id) {
+
+  $template_data = dosomething_mbp_get_template_data($application_id);
+
+  if (count($template_data) > 0) {
+    foreach ($template_data as $template_slug => $template_details) {
+      $mandrill_templates['select'][$template_slug] = $template_details['name'];
+    }
+  }
+  else {
+    $mandrill_templates = -1;
+  }
+
+  return $mandrill_templates;
+}
+
+/**
+ * Gather Mandrill template details with the label specific to the affiliate
+ * (target_ID).
+ *
+ * @parm string $application_id
+ *   Defined in message_broker_producer, a unique identifier of the site.
+ */
+function dosomething_mbp_get_template_data($application_id) {
```

Comment:
Also getting rid of this per https://github.com/DoSomething/dosomething/pull/3640#issuecomment-70723782